diff --git a/src/core/web_engine_library_info.cpp b/src/core/web_engine_library_info.cpp
--- a/src/core/web_engine_library_info.cpp
+++ b/src/core/web_engine_library_info.cpp
@@ -232,8 +232,9 @@ QString localesPath()
 #if defined(Q_OS_DARWIN) && defined(QT_MAC_FRAMEWORK_BUILD)
             candidatePaths << getResourcesPath(frameworkBundle()) % QDir::separator()
                             % "qtwebengine_locales"_L1;
 #endif
+            candidatePaths << QStringLiteral("/app/translations/qtwebengine_locales");
             for (auto &p : QLibraryInfo::paths(QLibraryInfo::TranslationsPath))
                 candidatePaths << p % QDir::separator() % "qtwebengine_locales"_L1;
             candidatePaths << fallbackDir();
         }
@@ -298,8 +299,9 @@ QString dictionariesPath(bool showWarnings)
             QString frameworkDictionariesPath =
                     getResourcesPath(frameworkBundle()) % "/qtwebengine_dictionaries"_L1;
             candidatePaths.append(std::move(frameworkDictionariesPath));
 #endif
+            candidatePaths << QStringLiteral("/app/qtwebengine_dictionaries");
             for (auto &p : QLibraryInfo::paths(QLibraryInfo::DataPath))
                 candidatePaths.append(p % QDir::separator() % "qtwebengine_dictionaries"_L1);
         }
 
@@ -353,8 +355,9 @@ QString resourcesPath()
             includeOverrideMessage = true;
 #if defined(Q_OS_DARWIN) && defined(QT_MAC_FRAMEWORK_BUILD)
             candidatePaths << getResourcesPath(frameworkBundle());
 #endif
+            candidatePaths << QStringLiteral("/app/resources");
             for (auto &p : QLibraryInfo::paths(QLibraryInfo::DataPath)) {
                 candidatePaths << p % QDir::separator() % "resources"_L1;
                 candidatePaths << p;
             }
